packages/main/cypress/specs/DateRangePicker.cy.tsx

```diff
@@ -0,0 +1,570 @@
+			.as("input")
+			.realClick()
+			.realPress("End")
+			.realPress(["Shift", "PageUp"]);
+
+		cy.get("@dateRangePicker")
+			.should("have.attr", "value", "Jul 16, 2020 @ Aug 29, 2020");
+
+		cy.get("@input")
+			.realPress(["Shift", "PageDown"]);
+
+		cy.get("@dateRangePicker")
+			.should("have.attr", "value", "Jul 16, 2020 @ Jul 29, 2020");
+
+		cy.get("@input")
+			.realPress("Home")
+			.realPress(["Shift", "PageDown"]);
+
+		cy.get("@dateRangePicker")
+			.should("have.attr", "value", "Jun 16, 2020 @ Jul 29, 2020");
+
+		cy.get("@input")
+			.realPress(["Shift", "PageUp"]);
+
+		cy.get("@dateRangePicker")
+			.should("have.attr", "value", "Jul 16, 2020 @ Jul 29, 2020");
+	});
+
+	it("Page up/down increments/decrements the year value", () => {
+		cy.mount(<DateRangePicker
+			formatPattern="MMM d, y"
+			value="Jul 16, 2020 @ Jul 29, 2020"
+			delimiter="@">
+		</DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.as("input")
+			.realClick()
+			.realPress("End")
+			.realPress(["Control", "Shift", "PageUp"]);
+
+		cy.get("@dateRangePicker")
+			.should("have.attr", "value", "Jul 16, 2020 @ Jul 29, 2021");
+
+		cy.get("@input")
+			.realPress(["Control", "Shift", "PageDown"]);
+
+		cy.get("@dateRangePicker")
+			.should("have.attr", "value", "Jul 16, 2020 @ Jul 29, 2020");
+
+		cy.get("@input")
+			.realPress("Home")
+			.realPress(["Control", "Shift", "PageDown"]);
+
+		cy.get("@dateRangePicker")
+			.should("have.attr", "value", "Jul 16, 2019 @ Jul 29, 2020");
+
+		cy.get("@input")
+			.realPress(["Control", "Shift", "PageUp"]);
+
+		cy.get("@dateRangePicker")
+			.should("have.attr", "value", "Jul 16, 2020 @ Jul 29, 2020");
+	});
+
+	it("Enter keyboard key confirms the date range in the input field", () => {
+		cy.mount(<DateRangePicker formatPattern="MMM d, y" delimiter="@"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realType("Jul 17, 2020 @ Jul 16, 2020")
+			.realPress("Enter");
+
+		cy.get("@dateRangePicker")
+			.should("have.attr", "value", "Jul 16, 2020 @ Jul 17, 2020");
+	});
+
+	it("Focus out of the input field confirms the date range", () => {
+		cy.mount(<DateRangePicker formatPattern="MMM d, y" delimiter="@"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realType("Jul 17, 2020 @ Jul 16, 2020")
+			.realPress("Tab");
+
+		cy.get("@dateRangePicker")
+			.should("have.attr", "value", "Jul 16, 2020 @ Jul 17, 2020");
+	});
+
+	it("Delimiter is part of the format pattern", () => {
+		cy.mount(<DateRangePicker formatPattern="yyyy-MM-dd"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
```

Comment:
i think it will be good to ensure that the input is focused after the click and before we type